datafusion/core/tests/sqllogictests/test_files/copy.slt

```diff
@@ -16,29 +16,141 @@
 # under the License.
 
 # tests for copy command
-
 statement ok
 create table source_table(col1 integer, col2 varchar) as values (1, 'Foo'), (2, 'Bar');
 
-# Copy from table
-statement error DataFusion error: This feature is not implemented: `COPY \.\. TO \.\.` statement is not yet supported
-COPY source_table  to '/tmp/table.parquet';
+# Copy to directory as multiple files
+query IT
+COPY source_table TO 'tests/sqllogictests/test_files/scratch/table' (format parquet, per_thread_output true);
+----
+2
+
+#Explain copy queries not currently working
```

Comment:
I noticed that `EXPLAIN <copy statement>` currently does not work. When prefixed by `EXPLAIN` the subsequent `COPY` token is being parsed into the `COPY ` statement defined in the `sqlparser` crate. We haven't implemented a logical plan and our expected syntax for `COPY` is different from `sqlparser`, so this leads to various errors.

When the `COPY ` token starts the statement, it is correctly parsed as a DFStatement defined within DataFusion.

Reply:
Here is where the parsing diverges. We might need to add some special case for parsing `explain copy`, but I'm not sure if there is a better way.

https://github.com/apache/arrow-datafusion/blob/6ad79165f6554a66aa5ed4c5d432401c2c162f69/datafusion/sql/src/parser.rs#L296-L331

Reply:
BTW I coded up support for EXPLAIN COPY in https://github.com/apache/arrow-datafusion/pull/7291